makefiles/tools/esptool.inc.mk

```
@@ -31,19 +32,38 @@ else
   FFLAGS += 0x10000 $(FLASHFILE).bin
 endif
 
-.PHONY: esp-image-convert esp-qemu
-# prepare image to flash: convert .elf to .bin
-esp-image-convert:
+# This target actually depends on $(FLASHFILE), but if we add $(FLASHFILE) here
+# it will cause it to be rebuilt even in "flash-only" mode because ELFFILE is
+# FORCEd. To avoid that we either add the dependency when calling make with
+# "all" or "flash", or we make it a PHONY target when only doing "flash-only".
+$(FLASHFILE).bin:
```

```suggestion
$(FLASHFILE).bin: $(FLASHFILE)
```

Or how about sticking to `ELFFILE` instead?

```suggestion
$(ELFFILE).bin: $(ELFFILE)
```

---

Strictly speaking, I think `FLASHFILE` should be `$(ELFFILE).bin`. I made that change in a second fixup commit now.